ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/PushPhysicalPlanBuilder.java

```diff
@@ -0,0 +1,169 @@
+      } else if (currentLogicalNode instanceof PullFilterNode) {
+        currentPhysicalOp = translateFilterNode((PullFilterNode) currentLogicalNode);
+      } else if (currentLogicalNode instanceof DataSourceNode) {
+        currentPhysicalOp = translateDataSourceNode(
+            (DataSourceNode) currentLogicalNode);
+        dataSourceOperator = (PushDataSourceOperator) currentPhysicalOp;
+      } else {
+        throw new KsqlException(String.format(
+            "Error in translating logical to physical plan for scalable push queries:"
+                + " unrecognized logical node %s.", currentLogicalNode));
+      }
+
+      if (prevPhysicalOp == null) {
+        rootPhysicalOp = currentPhysicalOp;
+      } else {
+        prevPhysicalOp.addChild(currentPhysicalOp);
+      }
+      prevPhysicalOp = currentPhysicalOp;
+      // Exit the loop when a leaf node is reached
+      if (currentLogicalNode.getSources().isEmpty()) {
+        break;
+      }
+      if (currentLogicalNode.getSources().size() > 1) {
+        throw new KsqlException("Push queries do not support joins or nested sub-queries yet.");
+      }
+      currentLogicalNode = currentLogicalNode.getSources().get(0);
+    }
+
+    if (dataSourceOperator == null) {
+      throw new IllegalStateException("DataSourceOperator cannot be null in Push physical plan");
+    }
+    return new PushPhysicalPlan(
+        rootPhysicalOp,
+        (rootPhysicalOp).getLogicalNode().getSchema(),
+        queryId,
+        dataSourceOperator.getScalablePushRegistry(),
+        dataSourceOperator,
+        context);
+  }
+
+  private ProjectOperator translateProjectNode(final PullProjectNode logicalNode) {
+    final ProcessingLogger logger = processingLogContext
+        .getLoggerFactory()
+        .getLogger(
+            QueryLoggerUtil.queryLoggerName(
+                QueryType.PULL_QUERY, contextStacker.push("PROJECT").getQueryContext())
+        );
+
+    return new ProjectOperator(
+        logger,
+        logicalNode
+    );
+  }
+
+  private SelectOperator translateFilterNode(final PullFilterNode logicalNode) {
+    final ProcessingLogger logger = processingLogContext
+        .getLoggerFactory()
+        .getLogger(
+            QueryLoggerUtil.queryLoggerName(
+                QueryType.PULL_QUERY, contextStacker.push("SELECT").getQueryContext())
+        );
+    return new SelectOperator(logicalNode, logger);
+  }
+
+  private AbstractPhysicalOperator translateDataSourceNode(
+      final DataSourceNode logicalNode
+  ) {
+    final ScalablePushRegistry scalablePushRegistry =
+        persistentQueryMetadata.getScalablePushRegistry()
+        .orElseThrow(() -> new IllegalStateException("Scalable push registry cannot be found"));
+    return new PeekStreamOperator(scalablePushRegistry, logicalNode);
+  }
+
+  private QueryId uniqueQueryId() {
+    return new QueryId("query_" + System.currentTimeMillis());
```

Comment:
Can we make the query ID convey some information about the 'topology' of the query and make the ID richer?

Reply:
+1. We can include some data source context here.

Reply:
I changed the prefix to `SCALABLE_PUSH_QUERY_`.

The only types of queries that contain anything more to the query id are persistent queries because they're user visible in that case.  E.g. normal push queries are here: https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java#L103.  At least at the moment, for pull and push queries, QueryIds aren't used in any meaningful way.  I'm happy to try to pipe in some meaningful info like the source name if we'll use it somewhere.